lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -158,10 +160,25 @@ class ConvertAtenMmOp : public OpConversionPattern<AtenMmOp> {
                        loc, zeroFill.getType(), ValueRange{lhs, rhs}, zeroFill)
                    .getResult(0);
     } else {
+      auto accumulatorTensorType = getDefaultAccType(rewriter, resultDTy);
+      if (accumulatorTensorType != resultDTy) {
+        auto zeroFillType = zeroFill.getType().cast<RankedTensorType>();
+        auto zeroFillShape = zeroFillType.getShape();
+        llvm::SmallVector<int64_t> zeroFillShapeVector;
+        for (int i = 0; i < zeroFillType.getRank(); ++i)
+          zeroFillShapeVector.push_back(zeroFillShape[i]);
+        auto accumulatorRankedTensorType =
+            RankedTensorType::get(zeroFillShapeVector, accumulatorTensorType);
+        zeroFill = rewriter.create<arith::ExtFOp>(
+            loc, accumulatorRankedTensorType, zeroFill);
+      }
       matmul = rewriter
                    .create<linalg::MatmulOp>(loc, zeroFill.getType(),
                                              ValueRange{lhs, rhs}, zeroFill)
                    .getResult(0);
+      if (accumulatorTensorType != resultDTy) {
+        matmul = rewriter.create<arith::TruncFOp>(loc, newResultType, matmul);
```

Comment:
This assumes floating point matmul which is not guaranteed. It could be a non-quantized matmul.

Reply:
I do agree with you that this could be a non-quantized matmul and if it is then there's no downcast (wouldn't pass the check). I'm not sure I follow